### PR TITLE
feat: npt (NoPorts Tunnel) - noports but without ssh

### PR DESCRIPTION
**Pull request by gkc**
**What I did**

**CLI and Core**
- New client-side binary, `npt`, in sshnoports package, with corresponding new lib/src/npt in noports_core package
- New request handler in the daemon to handle `npt` requests
- Some restructuring of SshnpParams to enable reuse between sshnp and npt clients
- Removed obsolete code
- Always ping daemon for features. (But do it such that overall time-to-session is not increased, by sending ping request to the daemon first, then dealing with the rvd, and only checking for the ping response after finishing communication with the rvd
- Added new method to SshnpdChannel to enable easier reuse of the daemon feature check code
- Added new DaemonFeature, `supportsPortChoice`
- Some more renaming of things for clarity
- Removed client-side code dealing with legacy (reverse ssh, v2 / v3) behaviour
- Added support for multiple socket pairs per session, not just one

**GUI** 
- Changed pubspec to point noports_core to ../noports_core rather than published package
- Made changes as required by changes in noports_core

**How I did it**

See commits, and some annotations on "Files Changed" tab.

**How to verify it**
Manual testing. A separate PR is in progress to provided automated e2e tests.

### CODE REVIEW

File: packages/dart/noports_core/lib/src/sshnp/util/sshnpd_channel/sshnpd_default_channel.dart [hunk 1]
```diff
@@ -34,7 +34,11 @@ mixin SshnpdDefaultPayloadHandler on SshnpdChannel {
 
   @override
   Future<SshnpdAck> handleSshnpdPayload(AtNotification notification) async {
-    if (notification.value?.startsWith('{') ?? false) {
+    bool validResponse = notification.value?.startsWith('{') ?? false;
+    if (!validResponse) {
+      logger.shout('invalid daemon response: ${notification.value}');
+      return SshnpdAck.acknowledgedWithErrors;
```
Comment (gkc) [inline packages/dart/noports_core/lib/src/sshnp/util/sshnpd_channel/sshnpd_default_channel.dart:40]: Added this block for clarity in logs

File: packages/dart/noports_core/lib/src/sshnpd/sshnpd_impl.dart [hunk 2]
```diff
@@ -82,10 +85,11 @@ class SshnpdImpl implements Sshnpd {
     required this.sshClient,
     this.makeDeviceInfoVisible = false,
     this.addSshPublicKeys = false,
-    this.localSshdPort = DefaultArgs.localSshdPort,
+    this.localSshdPort = DefaultSshnpdArgs.localSshdPort,
```
Comment (gkc) [inline packages/dart/noports_core/lib/src/sshnpd/sshnpd_impl.dart:88]: localSshdPort is only relevant to the daemon now

**Review by cconstab — DISMISSED** [review]
tested to this point more to do :-)

**Comment by gkc** [comment]
> tested to this point more to do :-)

grace period now added in socket_connector package and taken up here in order to accommodate situations where a client connects a socket, then disconnects and then tries to connect again

**Review by cconstab — APPROVED** [review]
tested with many clients of RDP/SSH all worked as expected

**Pull request merged**